Using Light Function Built-In "LIGHT" causes "Fragment Program Compilation" to fail

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10 Pro, Version 1709, Godot 3.0beta.build1, Nvidia GeForce GTX 1060 

**Issue description:**
Create a spatial node with children: MeshInstance, DirectionalLight and WorldEnvironment, add a SphereMesh to the MeshInstance, Add a new ShaderMaterial to the MeshInstance. Add a new Shader and type the following into the shader:

shader_type spatial;

void light() {
	vec3 light = LIGHT;
	DIFFUSE_LIGHT = vec3(0.0);
}

The SphereMesh turns grey and a large white oval shape covers the 3D view. If I comment out the line that assigns "LIGHT" to the ve3 "light, it looks as it should, a black sphere.

I get this in the terminal:

ERROR: _display_error_with_code: SceneShaderGLES3: Fragment Program Compilation Failed:
0(448) : error C1008: undefined variable "m_LIGHT"

   At: drivers\gles3\shader_gles3.cpp:167
ERROR: ShaderGLES3::get_current_version: SceneShaderGLES3: Fragment Program Compilation Failed:
0(448) : error C1008: undefined variable "m_LIGHT"

   At: drivers\gles3\shader_gles3.cpp:470
ERROR: ShaderGLES3::get_current_version: Method/Function Failed, returning: 0
   At: drivers\gles3\shader_gles3.cpp:478
ERROR: ShaderGLES3::bind: Condition ' !version ' is true. returned: false
   At: drivers\gles3\shader_gles3.cpp:122
ERROR: ShaderGLES3::_get_uniform: Condition ' !version ' is true. returned: -1
   At: .\drivers/gles3/shader_gles3.h:376
ERROR: ShaderGLES3::_get_uniform: Condition ' !version ' is true. returned: -1
   At: .\drivers/gles3/shader_gles3.h:376
ERROR: ShaderGLES3::_get_uniform: Condition ' !version ' is true. returned: -1
   At: .\drivers/gles3/shader_gles3.h:376
ERROR: ShaderGLES3::_get_uniform: Condition ' !version ' is true. returned: -1
   At: .\drivers/gles3/shader_gles3.h:376
ERROR: ShaderGLES3::_get_uniform: Condition ' !version ' is true. returned: -1
   At: .\drivers/gles3/shader_gles3.h:376


**Steps to reproduce:**
Any time i use the Built-In "LIGHT" in the light function, along with either a DIFFUSE_LIGHT out or a SPECULAR_LIGHT out i get the same strange oval shape. If I do not have any out from the light function, It looks fine.

**Link to minimal example project:**
[New Game Project.zip](https://github.com/godotengine/godot/files/1546373/New.Game.Project.zip)

--- a/drivers/gles3/shader_compiler_gles3.cpp
+++ b/drivers/gles3/shader_compiler_gles3.cpp
@@ -812,6 +812,7 @@ ShaderCompilerGLES3::ShaderCompilerGLES3() {
 	//for light
 	actions[VS::SHADER_SPATIAL].renames["VIEW"] = "view";
 	actions[VS::SHADER_SPATIAL].renames["LIGHT_COLOR"] = "light_color";
+	actions[VS::SHADER_SPATIAL].renames["LIGHT"] = "light";
 	actions[VS::SHADER_SPATIAL].renames["ATTENUATION"] = "attenuation";
 	actions[VS::SHADER_SPATIAL].renames["DIFFUSE_LIGHT"] = "diffuse_light";
 	actions[VS::SHADER_SPATIAL].renames["SPECULAR_LIGHT"] = "specular_light";

